tests/test_read_backed_experimental.py

```diff
@@ -0,0 +1,190 @@
+from __future__ import annotations
+
+from importlib.util import find_spec
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import pytest
+from scipy import sparse
+
+from anndata._core.anndata import AnnData
+from anndata.experimental import read_backed
+from anndata.tests.helpers import (
+    AccessTrackingStore,
+    as_dense_dask_array,
+    assert_equal,
+    gen_adata,
+    gen_typed_df,
+)
+
+
+@pytest.fixture(
+    params=[sparse.csr_matrix, sparse.csc_matrix, np.array, as_dense_dask_array],
+    ids=["scipy-csr", "scipy-csc", "np-array", "dask_array"],
+    scope="session",
+)
+def mtx_format(request):
+    return request.param
+
+
+@pytest.fixture(params=["zarr", "h5ad"], scope="session")
+def dskfmt(request):
+    return request.param
+
+
+@pytest.fixture(scope="session")
+def adata_remote_orig(
+    tmp_path_factory, dskfmt: str, mtx_format
+) -> tuple[AnnData, AnnData]:
+    orig_path = Path(tmp_path_factory.mktemp(f"orig.{dskfmt}"))
+    orig = gen_adata((1000, 1000), mtx_format)
+    orig.write_zarr(orig_path)
+    remote = read_backed(orig_path)
+    return remote, orig
+
+
+@pytest.fixture
+def adata_remote_with_store_tall_skinny(tmp_path_factory, mtx_format):
+    orig_path = Path(tmp_path_factory.mktemp("orig.zarr"))
+    M = 1000000  # forces zarr to chunk `obs` columns multiple ways - that way 1 access to `int64` below is actually only one access
+    N = 5
+    obs_names = pd.Index(f"cell{i}" for i in range(M))
+    var_names = pd.Index(f"gene{i}" for i in range(N))
+    obs = gen_typed_df(M, obs_names)
+    var = gen_typed_df(N, var_names)
+    orig = AnnData(
+        obs=obs,
+        var=var,
+        X=mtx_format(np.random.binomial(100, 0.005, (M, N)).astype(np.float32)),
+    )
+    orig.write_zarr(orig_path)
+    store = AccessTrackingStore(orig_path)
+    remote = read_backed(store)
+    return remote, store
+
+
+needs_xarray = pytest.mark.skipif(
+    not find_spec("xarray"), reason="Xarray is not installed"
+)
```

Comment:
you could just set [pytestmark](https://docs.pytest.org/en/7.1.x/reference/reference.html?highlight=pytest_marks#globalvar-pytestmark) once at the top, then the whole file is skipped and you don’t have to annotate each individual function.

```py
pytestmark = pytest.mark.skipif(
    not find_spec("xarray"), reason="Xarray is not installed"
)
```